blues/simulation.py

```diff
@@ -0,0 +1,284 @@
+        return stateinfo
+
+    def getWorkInfo(self, nc_integrator, parameters):
+        """Function that obtains the work and energies from the NCMC integrator.
+
+        Returns a dict of the specified parameters.
+
+        Parameters
+        ----------
+        nc_integrator : openmm.Context.Integrator
+            The integrator from the NCMC Context
+        parameters : list
+            list containing strings of the values to get from the integrator.
+            Default : ['total_work', 'lambda', 'shadow_work',
+                       'protocol_work', 'Eold', 'Enew','Epert']
+        """
+        workinfo = {}
+        for param in parameters:
+            workinfo['param'] = nc_integrator.getGlobalVariableByName(param)
+        return workinfo
+
+    def writeFrame(self, simulation, outfname):
+        """Extracts a ParmEd structure and writes the frame given
+        an OpenMM Simulation object"""
+        topology = simulation.topology
+        system = simulation.context.getSystem()
+        state = simulation.context.getState(getPositions=True,
+                                            getVelocities=True,
+                                            getParameters=True,
+                                            getForces=True,
+                                            getParameterDerivatives=True,
+                                            getEnergy=True,
+                                            enforcePeriodicBox=True)
+
+
+        # Generate the ParmEd Structure
+        structure = parmed.openmm.load_topology(topology, system,
+                                   xyz=state.getPositions())
+
+        structure.save(outfname,overwrite=True)
+        print('Saving Frame to', outfname)
+
+    def chooseMove(self):
+        """Function that chooses to accept or reject the proposed move.
+        """
+        md_state0 = self.current_state['md']['state0']
+        nc_state0 = self.current_state['nc']['state0']
+        nc_state1 = self.current_state['nc']['state1']
+
+        log_ncmc = self.nc_integrator.getLogAcceptanceProbability(self.nc_context)
+        randnum =  math.log(np.random.random())
+
+        # Compute Alchemical Correction Term
+        if np.isnan(log_ncmc) == False:
+            self.alch_sim.context.setPositions(nc_state1['positions'])
+            alch_state1 = self.getStateInfo(self.alch_sim.context, self.state_keys)
+            self.setSimState('alch', 'state1', alch_state1)
+            correction_factor = (nc_state0['potential_energy'] - md_state0['potential_energy'] + alch_state1['potential_energy'] - nc_state1['potential_energy']) * (-1.0/self.nc_integrator.kT)
+            log_ncmc = log_ncmc + correction_factor
+
+        if log_ncmc > randnum:
+            self.accept += 1
+            print('NCMC MOVE ACCEPTED: log_ncmc {} > randnum {}'.format(log_ncmc, randnum) )
+            #print('accCounter', float(self.accept)/float(stepsdone+1), self.accept)
```

Comment:
Remove old comments? Otherwise use a debug/logging option so they can be used if needed?